Source/Session.swift

```diff
@@ -490,12 +526,30 @@ extension Session: RequestDelegate {
         return session.configuration
     }
 
-    public func willRetryRequest(_ request: Request) -> Bool {
-        return retrier(for: request) != nil
+    public func willAttemptToRetryRequest(_ request: Request) -> Bool {
+        let willExecuteRetryPolicies = !retryPolicies(for: request).isEmpty
+        let willExecuteRetrier = retrier(for: request) != nil
+
+        return willExecuteRetryPolicies || willExecuteRetrier
     }
 
     public func retryRequest(_ request: Request, ifNecessaryWithError error: Error) {
-        guard let retrier = retrier(for: request) else { return }
+        for retryPolicy in retryPolicies(for: request) {
+            let (shouldRetry, timeDelay) = retryPolicy.shouldRetry(request, with: error)
+
+            if shouldRetry {
+                self.rootQueue.after(timeDelay) {
+                    guard !request.isCancelled else { return }
+
+                    request.requestIsRetrying()
+                    self.perform(request)
+                }
+
+                return
+            }
+        }
+
+        guard let retrier = retrier(for: request) else { request.finish(); return }
```

Comment:
I love how simple this ended up being in the new AF5 structure. I just added a small section here before the retrier that runs through the retry policies.